File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/score/ScoreDTO.java
package es.jocasolo.competitiveeventsapi.dto.score;

import java.io.Serializable;
import java.util.Date;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.dto.image.ImageDTO;
import es.jocasolo.competitiveeventsapi.dto.user.UserLiteDTO;
import es.jocasolo.competitiveeventsapi.enums.score.ScoreStatusType;
import lombok.Getter;
import lombok.Setter;

public class ScoreDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	private Integer id;

	@Getter
	@Setter
	private String value;

	@Getter
	@Setter
	private Date date;

	@Getter
	@Setter
	private ScoreStatusType status;

	@Getter
	@Setter
	private ImageDTO image;
	
	@Getter
	@Setter
	private UserLiteDTO user;

	@Override
	public String toString() {
		return String.format("ScoreDTO [id=%s, value=%s]", id, value);
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/score/ScoreDTO.kt
package es.jocasolo.competitiveeventsapp.dto.score

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.dto.HistoryItemDTO
import es.jocasolo.competitiveeventsapp.dto.image.ImageDTO
import es.jocasolo.competitiveeventsapp.dto.user.UserDTO
import es.jocasolo.competitiveeventsapp.enums.score.ScoreStatusType
import java.util.*

class ScoreDTO(
    @SerializedName("id") var id: String,
    @SerializedName("value") var value: String?,
    @SerializedName("date") var date: Date?,
    @SerializedName("status") var status: ScoreStatusType?,
    @SerializedName("image") var image: ImageDTO?,
    @SerializedName("user") var user: UserDTO?
) : HistoryItemDTO() {
}